crates/ere-zisk/src/lib.rs

```diff
@@ -24,6 +25,10 @@ include!(concat!(env!("OUT_DIR"), "/name_and_sdk_version.rs"));
 mod compile;
 mod error;
 
+/// It panics if `EreZisk::prove` is called concurrently, so we need a lock here
+/// to avoid that.
+static PROVE_LOCK: Mutex<()> = Mutex::new(());
```

Comment:
Add a lock here so we can run the test without needing the flag `--test-threads=1` or worrying about the prove method gets called concurrently.